assets/app/scripts/directives/logViewer.js

```diff
@@ -103,6 +150,24 @@ angular.module('openshiftConsole')
                 DataService.createStream($scope.kind, $scope.name, $scope.context, options);
 
               var lastLineNumber = 0;
+
+              var addLine = function(lineNumber, text) {
+                bufferedLines.push({
```

Comment:
this is the sort of intermediate allocation we could avoid by appending to the document fragment here